crates/modelardb_compression/src/types.rs

```diff
@@ -0,0 +1,622 @@
+/* Copyright 2023 The ModelarDB Contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+//! The types used throughout the crate.
```

Comment:
While it would be nice if all modules were organized the same, I do not think we can create a one-size-fits-all structure that works well for all module as I think it depends on the contents of each module. For `compression` there is a clear connection between the types as each either produce the next or is used as input to the next. In addition, the complete `types.rs` is currently only 614 lines. So for `compression` I think it currently makes sense to store the types in one module to make it simple  to understand how they are connected. For `storage` the types and functions are already clearly grouped into uncompressed and compressed, so I think moving the two different types of buffers into one file would make the module more complicated. We could maybe store the manager and buffer of each type in their own file, e.g., merging `uncompressed_data_manager.rs` and `uncompressed_data_buffer.rs`, however these files are already 1020 lines and 599 lines respectively, so the resulting files would be fairly big. Would do you think @chrthomsen?

Reply:
Good question... Given that one size does not fit all and that there seems to be doubt if a change would make `storage` simpler, I'm in favor of leaving it unchanged until it is (reasonably) clear that it will be a change to the better.